### ISSUE TITLE
A network of 66 winrate for lz -247

### ISSUE THREAD

**Issue by yehud** [leela-zero/leela-zero#2509]
With the permission of the author, I publish some of its information here 
   Using minigo v17’ s sgfs(60w)and leelazero's trainingdata(10w)
   We have tested 200 games
with lz-247 B:57:43 W:75:25(1000 playouts)
   This tells us maybe we can add some v17 data to the official network training.
   https://drive.google.com/open?id=1iiq4S_1mtSIpoK2CKYuJN5WWaZs1oZn4
    We want to test it officially on  lz's website, thank you!

**Comment by yehud** [comment]
1w=10thousand

**Comment by l1t1** [comment]
but lz cannot use minigo weight..

**Comment by l1t1** [comment]
someone tested it, it can be used by lz 0.17

**Comment by bubblesld** [comment]
The file size is more than 200mb. I tried to use quantize.py in lz, but not working in this format. I did some test matches with 1600 visits and other setting as the official test. This new net is 108:92 over v247. Anyone who is interested can check the result [here](https://drive.google.com/open?id=1CkfNIpyETN2gzDoPLY3FsJiwRnRYNNG1)

**Comment by amj** [comment]
@yehud This is pretty cool!  How'd you decide on the 6:1 ratio?  If i understand correctly, this is not using minigo's weights to do selfplay, but actually using our training data, correct?

**Comment by l1t1** [comment]
i use `python quantize_weights.py 768.txt` on the file, it works and shows
`Weight file difference L2-norm: 0.0542603191636`

the file size reduece by a half, and can load by leelaz 0.17
```
629,823,577 768.txt
357,106,495 768_quantized.txt

D:\tool\leela15>leelaz -w \768_quantized.txt -g
Using OpenCL batch size of 5
Using 10 thread(s).
RNG seed: 4609823530461688885
BLAS Core: Haswell
Detecting residual layers...v1...256 channels...40 blocks.
Initializing OpenCL (autodetecting precision).
Using OpenCL half precision (at least 5% faster than single).
Setting max tree size to 3736 MiB and cache size to 415 MiB.
genmove b
Thinking at most 36.0 seconds...
NN eval=0.451229
Playouts: 53, Win: 45.48%, PV: C16 Q17 D4 Q4 R15
13.9 average depth, 34 max depth
2299 non leaf nodes, 1.18 average children
2723 visits, 947627 nodes, 2722 playouts, 73 n/s

= Q17

NN eval=0.536836
17.6 average depth, 47 max depth
1641 non leaf nodes, 1.18 average children

1941 visits, 666281 nodes
```

**Comment by l1t1** [comment]
upload the file  `768_quantized.txt.gz`
to https://files.fm/u/mmnweyk7
file size 93,113,781 bytes

**Comment by bubblesld** [comment]
yes, it works. I used old version of quantize.py. It works with the new version.

**Comment by l1t1** [comment]
maybe the  quantized version is weaker than the original. i remember the elfv2 is the original too.

**Comment by lcwd2** [comment]
how was this model trained?

**Comment by l1t1** [comment]
768 vs 768 quantized version
`validation -k 768 -n \768.txt -o "-g -v 100 -r 5 -w" -n \768_q.gz -o "-g -v 100 -r 5 -w"`

```
16 wins, 9 losses
25 games played.
Status: 0 LLR 0.598573 Lower Bound -2.94444 Upper Bound 2.94444
```

**Comment by yehud** [comment]
> @yehud This is pretty cool! How'd you decide on the 6:1 ratio? If i understand correctly, this is not using minigo's weights to do selfplay, but actually using our training data, correct?

at the begining,the ratio is 3:1,but the result isn't very good,6:1 is round 2

**Comment by yehud** [comment]
i feel confused about the result of todays test.We expect it to be significantly stronger than lz 247 in the offical test.we test it with the latest bjiyxo's network(40b-245c-32k)and get the result 119:81(1000po),we don't know why it only have 52% winrate over 247
here is the test(66winrate)
here is the test(59.5winrate) 
oh sorry ican't connected to google driver now ,i will up load test next time
we have 2080ti*1,training batchsize:32 lr:0.000025
btw,what is the learning rate and batchsize leelazero using now?

**Comment by yehud** [comment]
we using no quantized version

**Comment by bubblesld** [comment]
25 games with 100 visits is too few. I have provided 200 games with 1600 visits above.

The reason that we used the quantized version is that it caused problems before when the file size was too large.

**Comment by l1t1** [comment]
52 wins, 41 losses
93 games played.
Status: 0 LLR 0.638881 Lower Bound -2.94444 Upper Bound 2.94444

```
\768.txt v \768_q.g ( 102 games)
              wins        black       white
\768.txt   57 55.88%   26 56.52%   31 55.36%
\768_q.g   45 44.12%   20 43.48%   25 44.64%
                       46 45.10%   56 54.90%
102 games played.
Status: 0 LLR 0.693968 Lower Bound -2.94444 Upper Bound 2.94444
```

**Comment by l1t1** [comment]
set decimal 4 
```
python quantize_weights4.py 768.txt
Weight file difference L2-norm: 0.00552380046278

404,745,653 768_quantized.txt
120,039,033 768_4q.txt.gz
  93,113,781 768_q.gz
```
about 25% bigger than quantized decimal 3 weight gz file, but strength is better
```
\768.txt v \768_4q. ( 95 games)
              wins        black       white
\768.txt   51 53.68%   22 55.00%   29 52.73%
\768_4q.   44 46.32%   18 45.00%   26 47.27%
                       40 42.11%   55 57.89%
95 games played.
Status: 0 LLR 0.221195 Lower Bound -2.94444 Upper Bound 2.94444
```
https://files.fm/u/qnfqj42z

**Comment by bubblesld** [comment]
I ran 268 games with 1600 visits. 
Original vs quantized (the one in official website) = 131:141
I do not see statistically significant difference. The quantized one is in fact above 50% (probably due to luck).  

People who are interested in these test games, can check [here](https://drive.google.com/open?id=1wIxil3jyUHVJQIAgEBKjT9ON_nqK_bIF)

**Comment by yssaya** [comment]
I have tested 600 games by using opening book for first 10 moves. Black and white are alternate per a opening.

LZ-247 is 294 wins, 306 losses, against "swa-16-768000". Winrate is 0.49.(400 visit/move)

Test command is like this,
`$ java -jar ~/go/gogui-twogtp.jar -black "~/go/leelaz17 -v400 -r5 -g -q -d --noponder -w ~/networks/901e04ded97854dc028f574b77839c0b226e265914d68161c47ee9363b1c27fc" -white "~/go/leelaz17 -v400 -r5 -g -q -d --noponder -w ~/go/leelaz-model-swa-16-768000.gz" -games 2000 -size 19 -alternate -sgffile memo -auto -komi 7.5 -openings "~/go/opening/19x19_gogod_2014_1000"`
Opening book is
https://github.com/yssaya/leela-zero-ladder/tree/next/opening/19x19_gogod_2014_1000

**Comment by yehud** [comment]
thank you&nbsp;






------------------&nbsp;原始邮件&nbsp;------------------
发件人: "bubblesld"<notifications@github.com&gt;; 
发送时间: 2019年10月10日(星期四) 中午12:02
收件人: "leela-zero/leela-zero"<leela-zero@noreply.github.com&gt;; 
抄送: "雪落千寒"<1944844661@qq.com&gt;; "Mention"<mention@noreply.github.com&gt;; 
主题: Re: [leela-zero/leela-zero] A network of 66 winrate for lz -247 (#2509)




I ran 268 games with 1600 visits.
 Original vs quantized (the one in official website) = 131:141
 I do not see statistically significant difference. The quantized one is in fact above 50% (probably due to luck).
 
People who are interested in these test games, can check here

**Comment by bubblesld** [comment]
I ran another test with the original one (non-quantized) vs [c8ac3a70](http://zero.sjeng.org/networks/c8ac3a700e3ec046d392baa64dfeed26466bd8c017337a8817a2e4e3d8e8bc58.gz). The [result](https://drive.google.com/open?id=12dnxz9rkpW8qdEUq1umveMsg18MSFzTx) is pretty close at 197:203.

**Comment by l1t1** [comment]
```
C:\APPS\l0gpu17\validation.exe -k lz247-minigo17 -s "0:1" -g 5 -n C:\APPS\net\901e04de.gz -o "-g -v 2400 --gpu 0 --gpu 1 -r 5 -t 2 --batchsize 1 --noponder -q -d --timemanage off --precision single -w" -n C:\APPS\net\leelaz-model-swa-16-768000.gz -o "-g -v 2400 --gpu 0 --gpu 1 -r 5 -t 2 --batchsize 1 --noponder -q -d --timemanage off --precision single -w " -- C:\APPS\l0gpu17\leelaz -- C:\APPS\l0gpu17\leelaz

Code:
#247 v MG17 ( 400 games)
          wins        black       white
#247  178 44.50%   72 43.64%  106 45.11%
MG17  222 55.50%   93 56.36%  129 54.89%
                  165 41.25%  235 58.75%
400 games played.
Status: 0 LLR -0.128628 Lower Bound -2.94444 Upper Bound 2.94444
```
https://lifein19x19.com/viewtopic.php?p=249460#p249460
